### ISSUE TITLE
Webpack repo fails lint out of the box on windows

### ISSUE THREAD

**Issue by aboktor** [webpack/webpack#16992]
Lint fails on the CR in EOL of files on windows. I tried cloning the repo on linux and things worked as expected.
I tried with git's `core.autocrlf` set to `true` and set to `false` and they both failed. My analysis is that this happens because `.gitattributes` has `* text=auto` which is equivalent to having `autocrlf` to `true` all the time.
On windows, this means that EOLs will be CRLF, which fails lint.

## Options to solve this problem
- Set `* text eol=lf` in `.gitattributes`
- Add `*.js eol=lf` below `* text=auto` in `.gitattributes`

Talked to @TheLarkInn briefly about this issue.

**Comment by alexander-akait** [comment]
It is expected, `core.autocrlf` is an option for this (and you should use it), many of OSS repos use `* text=auto`:
- https://github.com/facebook/react/blob/main/.gitattributes#L1
- https://github.com/microsoft/vscode/blob/main/.gitattributes#L1

Can list more if you want, set `core.autocrlf` to `input`, your IDE should read `.editorconfig` (most of IDE can do it) and we have `end_of_line = lf`

**Comment by TheLarkInn** [comment]
I'm going to add a todo for myself add a note about this in our contributing guide. It should be there.

**Comment by Zlatkovsky** [comment]
For what it's worth, I too ran into this issue when cloning the webpack repo, and having Prettier immediately complain about the carriage returns:

![image](https://user-images.githubusercontent.com/2230453/232832256-ec6848a7-e108-45af-8c29-ac373b5efd87.png)

It was not obvious to me how to fix it (and fiddling with git settings didn't seem to help, though I'll admit it was my first time encountering something like this, so I may have missed the obvious). I eventually did find a solution, albeit a somewhat awkward one:

> 0. If `node_modules` is already created, delete it (e.g., `rimraf node_modules`). That will make the script run much faster.
> 1. Open the folder in Windows Explorer
> 2. Right click and open Git Bash
> 3. Run `find . -name "*.js" -exec dos2unix {} \;`
> 4. Commit the resulting file changes in my clone/branch/fork. (It's OK, they don't show up in the eventual PR into Webpack).

(Steps inspired by https://superuser.com/a/1252691)

It took several minutes for the script to scan through all the files, but it did do the trick once it was done.

@TheLarkInn , would definitely be good to have the definitive steps in the contributing guide, and hopefully simpler ones than what I ended up doing :-). And ideally, to have things "just work" out of the box if that's feasible.
